plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergConnectorSmokeTest.java

```diff
@@ -108,4 +113,40 @@ protected boolean isFileSorted(Location path, String sortColumnName)
     {
         return checkOrcFileSorting(fileSystem, path, sortColumnName);
     }
+
+    @Test
+    public void testRowConstructorColumnLimitForMergeQuery()
+    {
+        String[] colNames = {"orderkey", "custkey", "orderstatus", "totalprice", "orderpriority", "clerk", "shippriority", "comment", "orderdate"};
+        String[] colTypes = {"bigint", "bigint", "varchar", "decimal(12,2)", "varchar", "varchar", "int", "varchar", "date"};
+        String tableDefinition = "(";
+        String columns = "(";
+        String selectQuery = "select ";
+        String notMatchedClause = "";
+        String matchedClause = "";
+        // Creating merge query with 325 columns
+        for (int i = 0; i < 36; i++) {
+            for (int j = 0; j < 9; j++) {
+                tableDefinition += colNames[j] + "_" + i + " " + colTypes[j] + ",";
```

Comment:
Probably worth lifting the column name and type into a local variable here to help with readability, ie:
```
String columnName = colNames[j];
String columnType = colTypes[j];
```